packages/terra-tag/src/TagList.jsx

```diff
@@ -0,0 +1,331 @@
+        currentTag.current = tags[focusNode.current].id;
+        setTabIndex('0');
+        if (isRollUpRemoved.current) { // if the rollup tag was triggered with keyboard focus the node
+          focusCurrentNode();
+          isRollUpRemoved.current = false;
+        }
+      }
+    }
+  }, [children, isCollapsed, isCollapsible]);
+
+  const handleResize = useCallback((entries) => {
+    if (!Array.isArray(entries)) {
+      return;
+    }
+
+    setUpdatedCount(React.Children.count(children));
+    setRollUpCount(React.Children.count(children));
+
+    if (isCollapsible) {
+      generateRollUp();
+    }
+
+    resetTabIndex();
+    handleFocusOnRollUpTrigger();
+  }, [children, isCollapsible, resetTabIndex, handleFocusOnRollUpTrigger, generateRollUp]);
+
+  useLayoutEffect(() => {
+    let observer = new ResizeObserver((entries) => {
+      handleResize(entries);
+    });
+    observer.observe(filterTagsRef.current.parentNode);
+
+    return () => {
+      observer.disconnect();
+      observer = null;
+    };
+  }, [children, handleResize]);
+
+  const focusNextNode = (tags, rollUpTag) => {
+    const rollUpTagId = rollUpTag ? rollUpTag.getAttribute('id') : null;
+    if (focusNode.current + 1 <= tags.length) {
+      setTabIndex('-1');
+      // if the next tag is roll up tag, focus the roll up tag
+      if (rollUpTag && focusNode.current + 1 === tags.length) {
+        currentTag.current = rollUpTagId;
+      } else if (focusNode.current + 1 < tags.length) { // focus the next available tag
+        focusNode.current += 1;
+        currentTag.current = children[focusNode.current].props.id;
+      }
+      setTabIndex('0');
+      focusCurrentNode();
+    }
+  };
+
+  const focusPreviousNode = (tags, rollUpTag) => {
+    const rollUpTagId = rollUpTag ? rollUpTag.getAttribute('id') : null;
+    if (currentTag.current === rollUpTagId || focusNode.current >= 1) {
+      setTabIndex('-1');
+      if (rollUpTag && currentTag.current === rollUpTagId) { // If rollup tag, then focus the roll up tag
+        focusNode.current = tags.length - 1;
+        currentTag.current = children[focusNode.current].props.id;
+      } else {
+        focusNode.current -= 1;
+        currentTag.current = children[focusNode.current].props.id;
+      }
+      setTabIndex('0');
+      focusCurrentNode();
+    }
+  };
+
+  const handleTagListKeyDown = (event) => {
+    const tags = [...filterTagsRef.current.querySelectorAll('[data-terra-tag]')];
+    const rollUpTag = filterTagsRef.current.querySelector('[data-terra-rollup-tag]');
+
+    switch (event.keyCode) {
+      case KEY_RIGHT:
+        event.preventDefault();
+        focusNextNode(tags, rollUpTag);
+        break;
+      case KEY_LEFT:
+        event.preventDefault();
+        focusPreviousNode(tags, rollUpTag);
+        break;
+      case KEY_HOME:
+        event.preventDefault();
+        setTabIndex('-1');
+        focusNode.current = 0;
+        currentTag.current = tags[focusNode.current].id;
```

Comment:
Should we have refs on the Tags component elements instead of the element ids?

Reply:
Updated. Now have removed the support of ID.
Thanks